libs/msc-flash/msc-flash-harmony/src/lib/selectors/select-is-flashing-in-state.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { createSelector } from "@reduxjs/toolkit"
+import { FlashingProcessState } from "../constants"
+import { selectFlashingProcessState } from "./select-flashing-process-state"
+
+export const selectIsFlashingInState = (targetState: FlashingProcessState) =>
+  createSelector(
+    selectFlashingProcessState,
+    (flashingProcessState) => flashingProcessState === targetState
+  )
```

Comment:
I'm not sure this is needed. You can do the same by using the already existing selector `selectFlashingProcessState`. Creating a specialized selector for such comparison doesn't make sense unless you're going to use it hundreds of times and with the same `targetState` argument.

Notice how the second approach is much easier and more flexible to use:
```TSX
const isDownloadingFiles = useSelector((state: ReduxRootState) => { 
  return selectIsFlashingInState(FlashingProcessState.DownloadingFiles)
})
const isUnpackingFiles = useSelector((state: ReduxRootState) => { 
  return selectIsFlashingInState(FlashingProcessState.UnpackingFiles)
})

// ...

if (isDownloadingFiles) {
 // do something
} else if (isUnpackingFiles) {
  // do something
}
```
vs
```TSX
const flashingState = useSelector(selectFlashingProcessState)
const isDownloadingFiles = flashingState === FlashingProcessState.DownloadingFiles
const isUnpackingFiles = flashingState === FlashingProcessState.UnpackingFiles

if (isDownloadingFiles) {
 // do something
} else if (isUnpackingFiles) {
  // do something
}

// it also allows to use switch instead

switch (flashingState) {
  case FlashingProcessState.DownloadingFiles:
	// do something
	break
  case FlashingProcessState.UnpackingFiles:
	// do something
	break
}
```

Reply:
I created this selector mainly for memoization – when the Redux state doesn't change, the result is cached, avoiding unnecessary recalculations. Your approach is simpler, but my goal was to prevent extra computations.